Reuse rhpv3 transports instead of dealing a host for each operation

ChrisSchinnerl opened the pull request.

n8maninger commented:
Adding a 5-10 min keep-alive when the ref counter is 0 might help further reduce setup cost.

ChrisSchinnerl commented:
> Adding a 5-10 min keep-alive when the ref counter is 0 should help further reduce transport setup costs.

I added a commit that just uses `time.After` and reverted it again which adds something like that. I'll f/u with something but I feel like we need a way to detect whether a mux is broken before doing that. To avoid being stuck with a broken mux for 10 minutes or potentially forever.

peterjan commented on worker/rhpv3.go:
```diff
@@ -53,6 +53,113 @@ var (
+	transport := t.t
+	t.mu.Unlock()
+	if transport == nil {
+		return nil, errors.New("transport closed")
```
`nit` it makes sense to make this a variable I think

peterjan reviewed (approved):
looks great, excited to see if the benchmarks improve 🙇‍♂️
